File: 118_Pascal'sTriangle.py
class Solution(object):
    def generate(self, numRows):
        """
        Given numRows, generate the first numRows of Pascal's triangle.
        :type numRows: int
        :rtype: List[List[int]]
        """
        r = []
        for i in xrange(numRows):
            if i == 0:
                r.append([1])
            else:
                t = [1]
                for j in xrange(len(r[i-1]) - 1):
                    t.append(r[i-1][j] + r[i-1][j+1])
                t.append(1)
                r.append(t)
        
        return r

File: 118.Pascal'sTriangle.java
import java.util.ArrayList;

class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        ArrayList<Integer> row = new ArrayList<Integer>();
        for (int i = 0; i < numRows; i += 1) {
            row.add(0, 1);
            for (int j = 1; j < row.size() - 1; j += 1) {
                row.set(j, row.get(j) + row.get(j + 1));
            }
            result.add(new ArrayList<Integer>(row));
        }
        return result;
    }
}